Source/WebKit/UIProcess/API/mac/WKWebViewMac.mm

```diff
@@ -1177,6 +1195,26 @@ - (void)_web_didChangeContentSize:(NSSize)newSize
 {
 }
 
+- (void)_web_windowWillEnterFullScreen
+{
+#if HAVE(NSWINDOW_SNAPSHOT_READINESS_HANDLER)
+    if (!self.window || ![self.window respondsToSelector:@selector(_holdResizeSnapshotWithReason:)])
+        return;
+
+    _windowSnapshotReadinessHandler = makeBlockPtr([self.window _holdResizeSnapshotWithReason:@"web view entering full screen"]);
+#endif
```

Comment:
Nit - you could probably factor this code out into a helper function like `-_holdWindowResizeSnapshotWithReasion:(NSString *)reason;`, to avoid a tiny bit of code duplication.